src/mapper.rs

```diff
@@ -189,44 +189,253 @@ impl<'s> ProguardMapper<'s> {
     /// Returns zero or more [`StackFrame`]s, based on the information in
     /// the proguard mapping. This can return more than one frame in the case
     /// of inlined functions. In that case, frames are sorted top to bottom.
-    ///
-    /// [`StackFrame`]: struct.StackFrame.html
     pub fn remap_frame(&'s self, frame: &StackFrame<'s>) -> RemappedFrameIter<'s> {
-        if let Some(class) = self.classes.get(frame.class.as_ref()) {
-            if let Some(members) = class.members.get(frame.method.as_ref()) {
+        if let Some(class) = self.classes.get(frame.class) {
+            if let Some(members) = class.members.get(frame.method) {
                 let mut frame = frame.clone();
-                frame.class = class.original.into();
+                frame.class = class.original;
                 return RemappedFrameIter::members(frame, members.iter());
             }
         }
         RemappedFrameIter::empty()
     }
 
-    /// Remaps a complete Java StackTrace.
-    pub fn remap_stacktrace(&'s self, input: &'s str) -> Result<String, std::fmt::Error> {
+    /// Remaps a throwable which is the first line of a full stacktrace.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use proguard::{ProguardMapper,Throwable};
+    ///
+    /// let mapping = "com.example.Mapper -> a.b:";
+    /// let mapper = ProguardMapper::from(mapping);
+    ///
+    /// let throwable = Throwable::try_parse(b"a.b: Crash").unwrap();
+    /// let mapped = mapper.remap_throwable(&throwable);
+    ///
+    /// assert_eq!(Some(Throwable::with_message("com.example.Mapper", "Crash")), mapped);
+    /// ```
+    pub fn remap_throwable<'a>(&'a self, throwable: &Throwable<'a>) -> Option<Throwable<'a>> {
+        if let Some(class) = self.remap_class(throwable.class) {
+            Some(Throwable {
+                class,
+                message: throwable.message,
+            })
+        } else {
+            None
+        }
+    }
+
+    /// Remaps a complete Java StackTrace, similar to [`Self::remap_stacktrace`] but instead works on
+    /// strings as input and output.
+    pub fn remap_stacktrace(&self, input: &str) -> Result<String, std::fmt::Error> {
```

Comment:
```suggestion
    pub fn remap_stacktrace(&'s self, input: &'s str) -> Result<String, std::fmt::Error> {
```
I mean this shouldn’t change anything really, but lets keep the function signature exactly as it was before.

Reply:
Yeah this is exactly the same because lifetimes can be elided in this case. But we can put them back of course, no problem 👍